### PR TITLE
Dedupe volumes when specified twice in a config

### PR DESCRIPTION
**Pull request by solarkennedy**
I thought for sure docker would tolerate this.

Internal ticket PAASTA-7920.

### CODE REVIEW

File: paasta_tools/utils.py [hunk 1]
```diff
@@ -394,6 +394,11 @@ def get_net(self):
         """
         return self.config_dict.get('net', 'bridge')
 
+    def get_volumes(self, system_volumes):
+        volumes = system_volumes + self.get_extra_volumes()
+        deduped = {v['containerPath'] + v['hostPath']: v for v in volumes}.values()
```
Comment (mattmb) [inline paasta_tools/utils.py]: This seems pretty good. Although it doesn't dictate which config "wins".

One possible problem: say system config has `/thing` mounted to `/thing` and the service config has`/another` mounted to `/thing/sub` then we know that definitely wont work right? Can we catch that case to?

Comment (Rob-Johnson) [inline paasta_tools/utils.py]: mmb: sure it does - extra_volumes will always have priority

**Comment by solarkennedy** [comment]
Python 3 tests failing...

File: paasta_tools/utils.py (hunk 1 above)
Comment (solarkennedy) [inline paasta_tools/utils.py]: I've added a test to assert this and updated the docs.

**Pull request merged**